diff --git a/package.json b/package.json
index b46f9e11d988e11c89fbaf580e6bbc74502c6e7f..143135bd80304d991581643016799b0a0964c83b 100644
--- a/package.json
+++ b/package.json
@@ -16,8 +16,14 @@
     },
     "./zod": {
       "umd": "./zod/dist/zod.umd.js",
-      "import": "./zod/dist/zod.mjs",
-      "require": "./zod/dist/zod.js"
+      "import": {
+        "types": "./zod/dist/index.d.ts",
+        "default": "./zod/dist/zod.mjs"
+      },
+      "require": {
+        "types": "./zod/dist/index.d.ts",
+        "default": "./zod/dist/zod.js"
+      }
     },
     "./yup": {
       "umd": "./yup/dist/yup.umd.js",
diff --git a/zod/dist/index.d.ts b/zod/dist/index.d.ts
index 18dffb9fef1bf09e83d9ff5e3aec325cf562c780..b55eff2beb1311c50274f7040dd88102e5858faf 100644
--- a/zod/dist/index.d.ts
+++ b/zod/dist/index.d.ts
@@ -1,2 +1,2 @@
-export * from './zod';
-export * from './types';
+export * from './zod.js';
+export * from './types.js';
diff --git a/zod/dist/zod.d.ts b/zod/dist/zod.d.ts
index 839632e3d8495c00933dada6dd9f7a7d4003311a..3ad42563f00f8fb0a6d09c7234b66653b84561fe 100644
--- a/zod/dist/zod.d.ts
+++ b/zod/dist/zod.d.ts
@@ -1,2 +1,2 @@
-import type { Resolver } from './types';
+import type { Resolver } from './types.js';
 export declare const zodResolver: Resolver;