diff --git a/cjs/connectors/injectedConnector.js b/cjs/connectors/injectedConnector.js
index da8347dc2097d71a3145b0b2b0fc5f5112185020..42ef11541025660e09f001097b2d0e460a501c7e 100644
--- a/cjs/connectors/injectedConnector.js
+++ b/cjs/connectors/injectedConnector.js
@@ -19,6 +19,7 @@ class InjectedConnector extends events_1.EventEmitter {
             throw new Error('window is not defined. This should not have happened.');
         }
         this.windowProvider = connectorWindowProperty;
+        if(typeof this.windowProvider?.on !== 'function') return
         this.windowProvider?.on('connect', async ({ chainId }) => {
             await this.calcAccountData();
         });
diff --git a/connectors/injectedConnector.js b/connectors/injectedConnector.js
index 291db15c0816b98d5827b2e9aa998a1cdd2516c2..a3f8aea0e6dffa3ad08117517d72bc7e482fb3cd 100644
--- a/connectors/injectedConnector.js
+++ b/connectors/injectedConnector.js
@@ -16,6 +16,7 @@ export class InjectedConnector extends EventEmitter {
             throw new Error('window is not defined. This should not have happened.');
         }
         this.windowProvider = connectorWindowProperty;
+        if(typeof this.windowProvider?.on !== 'function') return
         this.windowProvider?.on('connect', async ({ chainId }) => {
             await this.calcAccountData();
         });
