diff --git a/dist/api/extractors/babel.js b/dist/api/extractors/babel.js
index 3282166cda9ea4173f3e6edb530b0fea1bb9e7d7..1f0f5045556745c19ee2738342fa13fe1ecd7507 100644
--- a/dist/api/extractors/babel.js
+++ b/dist/api/extractors/babel.js
@@ -120,6 +120,7 @@ function getBabelParserOptions(filename, parserOptions) {
         "importAttributes", // stage3
         "explicitResourceManagement", // stage3,
         "decoratorAutoAccessors", // stage3,
+        "deferredImportEvaluation", // stage3
     ];
     if ([/\.ts$/, /\.mts$/, /\.cts$/, /\.tsx$/].some((r) => filename.match(r))) {
         parserPlugins.push("typescript");
