diff --git a/Modal/Modal.d.ts b/Modal/Modal.d.ts
index a9ad08c406c944ec47dc9fe09e33507119bccbac..83fe2496ac959ab98cc8b89d5beb1439970a25a6 100644
--- a/Modal/Modal.d.ts
+++ b/Modal/Modal.d.ts
@@ -100,6 +100,7 @@ export interface ModalOwnProps {
    * so it's simply `document.body` most of the time.
    */
   container?: PortalProps['container'];
+  scrollContainer?: Element;
   /**
    * If `true`, the modal will not automatically shift focus to itself when it opens, and
    * replace it to the last focused element when it closes.
