diff --git a/Modal/Modal.js b/Modal/Modal.js
index fdabf43967e73a1f8b093262578c318d60174b60..cf67d848b76f54a48bce99483d34f9225a4de19d 100644
--- a/Modal/Modal.js
+++ b/Modal/Modal.js
@@ -29,7 +29,7 @@ var _Backdrop = _interopRequireDefault(require("../Backdrop"));
 
 var _jsxRuntime = require("react/jsx-runtime");
 
-const _excluded = ["BackdropComponent", "BackdropProps", "closeAfterTransition", "children", "component", "components", "componentsProps", "disableAutoFocus", "disableEnforceFocus", "disableEscapeKeyDown", "disablePortal", "disableRestoreFocus", "disableScrollLock", "hideBackdrop", "keepMounted", "theme"];
+const _excluded = ["BackdropComponent", "BackdropProps", "closeAfterTransition", "children", "component", "components", "componentsProps", "disableAutoFocus", "disableEnforceFocus", "disableEscapeKeyDown", "disablePortal", "disableRestoreFocus", "disableScrollLock", "scrollContainer", "hideBackdrop", "keepMounted", "theme"];
 
 function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
 
@@ -108,6 +108,7 @@ const Modal = /*#__PURE__*/React.forwardRef(function Modal(inProps, ref) {
     disablePortal = false,
     disableRestoreFocus = false,
     disableScrollLock = false,
+    scrollContainer,
     hideBackdrop = false,
     keepMounted = false,
     // eslint-disable-next-line react/prop-types
@@ -123,6 +124,7 @@ const Modal = /*#__PURE__*/React.forwardRef(function Modal(inProps, ref) {
     disablePortal,
     disableRestoreFocus,
     disableScrollLock,
+    scrollContainer,
     hideBackdrop,
     keepMounted
   };
diff --git a/esm/Modal/Modal.js b/esm/Modal/Modal.js
index a1e513add26d5e12528a920368b80bf4f9b665fa..3ff7061e0a919e211ce9ec2ba3228b5dcfeb748e 100644
--- a/esm/Modal/Modal.js
+++ b/esm/Modal/Modal.js
@@ -83,6 +83,7 @@ const Modal = /*#__PURE__*/React.forwardRef(function Modal(inProps, ref) {
     disablePortal = false,
     disableRestoreFocus = false,
     disableScrollLock = false,
+    scrollContainer,
     hideBackdrop = false,
     keepMounted = false,
     // eslint-disable-next-line react/prop-types
@@ -99,6 +100,7 @@ const Modal = /*#__PURE__*/React.forwardRef(function Modal(inProps, ref) {
     disablePortal,
     disableRestoreFocus,
     disableScrollLock,
+    scrollContainer,
     hideBackdrop,
     keepMounted
   };
diff --git a/legacy/Modal/Modal.js b/legacy/Modal/Modal.js
index 7ad41dcfbc20cd78a03f2e2bd6dce0ca3199161f..7e5db68feafa35b3384fc84b519c059d40e3b997 100644
--- a/legacy/Modal/Modal.js
+++ b/legacy/Modal/Modal.js
@@ -90,12 +90,13 @@ var Modal = /*#__PURE__*/React.forwardRef(function Modal(inProps, ref) {
       disableRestoreFocus = _props$disableRestore === void 0 ? false : _props$disableRestore,
       _props$disableScrollL = props.disableScrollLock,
       disableScrollLock = _props$disableScrollL === void 0 ? false : _props$disableScrollL,
+      scrollContainer = props.scrollContainer,
       _props$hideBackdrop = props.hideBackdrop,
       hideBackdrop = _props$hideBackdrop === void 0 ? false : _props$hideBackdrop,
       _props$keepMounted = props.keepMounted,
       keepMounted = _props$keepMounted === void 0 ? false : _props$keepMounted,
       theme = props.theme,
-      other = _objectWithoutProperties(props, ["BackdropComponent", "BackdropProps", "closeAfterTransition", "children", "component", "components", "componentsProps", "disableAutoFocus", "disableEnforceFocus", "disableEscapeKeyDown", "disablePortal", "disableRestoreFocus", "disableScrollLock", "hideBackdrop", "keepMounted", "theme"]);
+      other = _objectWithoutProperties(props, ["BackdropComponent", "BackdropProps", "closeAfterTransition", "children", "component", "components", "componentsProps", "disableAutoFocus", "disableEnforceFocus", "disableEscapeKeyDown", "disablePortal", "disableRestoreFocus", "disableScrollLock", "scrollContainer", "hideBackdrop", "keepMounted", "theme"]);
 
   var _React$useState = React.useState(true),
       exited = _React$useState[0],
@@ -109,6 +110,7 @@ var Modal = /*#__PURE__*/React.forwardRef(function Modal(inProps, ref) {
     disablePortal: disablePortal,
     disableRestoreFocus: disableRestoreFocus,
     disableScrollLock: disableScrollLock,
+    scrollContainer: scrollContainer,
     hideBackdrop: hideBackdrop,
     keepMounted: keepMounted
   };
diff --git a/modern/Modal/Modal.js b/modern/Modal/Modal.js
index 3ad7aa00d702e98651e1cf0c1c4edf3e5d7cb9ac..3914b81b2115b377bfed17bd9076f009cf21a50a 100644
--- a/modern/Modal/Modal.js
+++ b/modern/Modal/Modal.js
@@ -1,6 +1,6 @@
 import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
 import _extends from "@babel/runtime/helpers/esm/extends";
-const _excluded = ["BackdropComponent", "BackdropProps", "closeAfterTransition", "children", "component", "components", "componentsProps", "disableAutoFocus", "disableEnforceFocus", "disableEscapeKeyDown", "disablePortal", "disableRestoreFocus", "disableScrollLock", "hideBackdrop", "keepMounted", "theme"];
+const _excluded = ["BackdropComponent", "BackdropProps", "closeAfterTransition", "children", "component", "components", "componentsProps", "disableAutoFocus", "disableEnforceFocus", "disableEscapeKeyDown", "disablePortal", "disableRestoreFocus", "disableScrollLock", "scrollContainer", "hideBackdrop", "keepMounted", "theme"];
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import ModalUnstyled, { modalUnstyledClasses } from '@mui/base/ModalUnstyled';
@@ -81,6 +81,7 @@ const Modal = /*#__PURE__*/React.forwardRef(function Modal(inProps, ref) {
     disablePortal = false,
     disableRestoreFocus = false,
     disableScrollLock = false,
+    scrollContainer,
     hideBackdrop = false,
     keepMounted = false,
     // eslint-disable-next-line react/prop-types
@@ -97,6 +98,7 @@ const Modal = /*#__PURE__*/React.forwardRef(function Modal(inProps, ref) {
     disablePortal,
     disableRestoreFocus,
     disableScrollLock,
+    scrollContainer,
     hideBackdrop,
     keepMounted
   };