diff --git a/index.js b/index.js
index 33778b5539b7fcd7a1e99474a4ecb1745fdfe508..153aa654f1c84cd6f6e817141146704b5ea74e61 100644
--- a/index.js
+++ b/index.js
@@ -8,10 +8,5 @@ module.exports = inquire;
  * @returns {?Object} Required module if available and not empty, otherwise `null`
  */
 function inquire(moduleName) {
-    try {
-        var mod = eval("quire".replace(/^/,"re"))(moduleName); // eslint-disable-line no-eval
-        if (mod && (mod.length || Object.keys(mod).length))
-            return mod;
-    } catch (e) {} // eslint-disable-line no-empty
     return null;
 }