diff --git a/dist/cjs/src/detector.js b/dist/cjs/src/detector.js
index 6a864611431c83c2aeb2628a57a3478cfbfbc4ff..c27fe0cbb7bdb7a15599514fb09cbafa7464d3ff 100644
--- a/dist/cjs/src/detector.js
+++ b/dist/cjs/src/detector.js
@@ -752,6 +752,13 @@ class Detector {
             return yield checkIsInBlacklist("domains", domain.host);
         });
     }
+
+    checkAddressInBlacklist(address) {
+        return __awaiter(this, void 0, void 0, function* () {
+            return yield checkIsInBlacklist("address", address.toLowerCase());
+        });
+    }
+
     checkSiteStatus(url) {
         return __awaiter(this, void 0, void 0, function* () {
             const domain = getTopDomainFromUrl(url);
diff --git a/dist/esnext/src/detector.js b/dist/esnext/src/detector.js
index a9649f6182d991e8359f82583c8f976ce403c316..6e354448aa82b1e4449cb8e860d95ed9a21c9caf 100644
--- a/dist/esnext/src/detector.js
+++ b/dist/esnext/src/detector.js
@@ -725,6 +725,9 @@ export class Detector {
             return false;
         return await checkIsInBlacklist("domains", domain.host);
     }
+    async checkAddressInBlacklist(address) {
+        return await checkIsInBlacklist("address", address.toLowerCase());
+    }
     async checkSiteStatus(url) {
         const domain = getTopDomainFromUrl(url);
         if (!domain || domain === null)
diff --git a/src/detector.ts b/src/detector.ts
index 45bb6c9c0fde9b7e9fe245e4c8691703b56425a5..f9b7498cf850e3359aa2df6ce7be85ec027a75ca 100644
--- a/src/detector.ts
+++ b/src/detector.ts
@@ -889,6 +889,10 @@ export class Detector {
     if (domain.host === null) return false;
     return await checkIsInBlacklist("domains", domain.host);
   }
+  
+  async checkAddressInBlacklist(address: string): Promise<Boolean> {
+    return await checkIsInBlacklist("address", address.toLowerCase());
+  }
 
   async checkSiteStatus(url: string): Promise<any> {
     const domain = getTopDomainFromUrl(url);
