diff --git a/cjs/integrations/functiontostring.js b/cjs/integrations/functiontostring.js
index cc7c8b4192f29d31a0e373427693f81351ad899f..91228d0bc5dacd0f40d4fcc3db207d8327ecc9ba 100644
--- a/cjs/integrations/functiontostring.js
+++ b/cjs/integrations/functiontostring.js
@@ -23,11 +23,13 @@ class FunctionToString  {constructor() { FunctionToString.prototype.__init.call(
     // eslint-disable-next-line @typescript-eslint/unbound-method
     originalFunctionToString = Function.prototype.toString;
 
+    try {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     Function.prototype.toString = function ( ...args) {
       const context = utils.getOriginalFunction(this) || this;
       return originalFunctionToString.apply(context, args);
     };
+    } catch {}
   }
 } FunctionToString.__initStatic();
 
diff --git a/esm/integrations/functiontostring.js b/esm/integrations/functiontostring.js
index ef0ca57ea6fce07660af45d9aa385092e466c69f..a3ebd036f1e15375e2222d3e47be5e2d440f38d2 100644
--- a/esm/integrations/functiontostring.js
+++ b/esm/integrations/functiontostring.js
@@ -21,11 +21,13 @@ class FunctionToString  {constructor() { FunctionToString.prototype.__init.call(
     // eslint-disable-next-line @typescript-eslint/unbound-method
     originalFunctionToString = Function.prototype.toString;
 
+    try {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     Function.prototype.toString = function ( ...args) {
       const context = getOriginalFunction(this) || this;
       return originalFunctionToString.apply(context, args);
     };
+    } catch {}
   }
 } FunctionToString.__initStatic();
 