diff --git a/dist/index.d.ts b/dist/index.d.ts
index 69a4b5419833c62f52fa6d07dd3b287c1cb12fd0..64a033158ccbc38bead38cc7d7fddd979d0b50c9 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,5 +1,5 @@
-import Client712 from './sign';
-import Client from './client';
+import Client712 from "./sign/index.js";
+import Client from "./client.js";
 declare const _default: {
     Client: typeof Client;
     Client712: typeof Client712;
@@ -409,42 +409,42 @@ declare const _default: {
         };
     };
     utils: {
-        call: typeof import("./utils").call;
-        multicall: typeof import("./utils").multicall;
-        subgraphRequest: typeof import("./utils").subgraphRequest;
-        ipfsGet: typeof import("./utils").ipfsGet;
-        getUrl: typeof import("./utils").getUrl;
-        getJSON: typeof import("./utils").getJSON;
-        sendTransaction: typeof import("./utils").sendTransaction;
-        getScores: typeof import("./utils").getScores;
-        getVp: typeof import("./utils").getVp;
-        validateSchema: typeof import("./utils").validateSchema;
-        getEnsTextRecord: typeof import("./utils").getEnsTextRecord;
-        getSpaceUri: typeof import("./utils").getSpaceUri;
-        getDelegatesBySpace: typeof import("./utils").getDelegatesBySpace;
-        clone: typeof import("./utils").clone;
-        sleep: typeof import("./utils").sleep;
-        getNumberWithOrdinal: typeof import("./utils").getNumberWithOrdinal;
+        call: typeof import("./utils.js").call;
+        multicall: typeof import("./utils.js").multicall;
+        subgraphRequest: typeof import("./utils.js").subgraphRequest;
+        ipfsGet: typeof import("./utils.js").ipfsGet;
+        getUrl: typeof import("./utils.js").getUrl;
+        getJSON: typeof import("./utils.js").getJSON;
+        sendTransaction: typeof import("./utils.js").sendTransaction;
+        getScores: typeof import("./utils.js").getScores;
+        getVp: typeof import("./utils.js").getVp;
+        validateSchema: typeof import("./utils.js").validateSchema;
+        getEnsTextRecord: typeof import("./utils.js").getEnsTextRecord;
+        getSpaceUri: typeof import("./utils.js").getSpaceUri;
+        getDelegatesBySpace: typeof import("./utils.js").getDelegatesBySpace;
+        clone: typeof import("./utils.js").clone;
+        sleep: typeof import("./utils.js").sleep;
+        getNumberWithOrdinal: typeof import("./utils.js").getNumberWithOrdinal;
         voting: {
-            'single-choice': typeof import("./voting/singleChoice").default;
-            approval: typeof import("./voting/approval").default;
-            quadratic: typeof import("./voting/quadratic").default;
-            'ranked-choice': typeof import("./voting/rankedChoice").default;
-            weighted: typeof import("./voting/weighted").default;
-            basic: typeof import("./voting/singleChoice").default;
+            "single-choice": typeof import("./voting/singleChoice.js").default;
+            approval: typeof import("./voting/approval.js").default;
+            quadratic: typeof import("./voting/quadratic.js").default;
+            "ranked-choice": typeof import("./voting/rankedChoice.js").default;
+            weighted: typeof import("./voting/weighted.js").default;
+            basic: typeof import("./voting/singleChoice.js").default;
         };
-        getProvider: typeof import("./utils/provider").default;
-        signMessage: typeof import("./utils/web3").signMessage;
-        getBlockNumber: typeof import("./utils/web3").getBlockNumber;
-        Multicaller: typeof import("./utils/multicaller").default;
-        getSnapshots: typeof import("./utils/blockfinder").getSnapshots;
+        getProvider: typeof import("./utils/provider.js").default;
+        signMessage: typeof import("./utils/web3.js").signMessage;
+        getBlockNumber: typeof import("./utils/web3.js").getBlockNumber;
+        Multicaller: typeof import("./utils/multicaller.js").default;
+        getSnapshots: typeof import("./utils/blockfinder.js").getSnapshots;
         validations: {
-            basic: typeof import("./validations/basic").default;
-            aave: typeof import("./validations/aave").default;
-            nouns: typeof import("./validations/nouns").default;
+            basic: typeof import("./validations/basic/index.js").default;
+            aave: typeof import("./validations/aave/index.js").default;
+            nouns: typeof import("./validations/nouns/index.js").default;
         };
-        getHash: typeof import("./sign/utils").getHash;
-        verify: typeof import("./sign/utils").verify;
+        getHash: typeof import("./sign/utils.js").getHash;
+        verify: typeof import("./sign/utils.js").verify;
         SNAPSHOT_SUBGRAPH_URL: {
             1: string;
             4: string;
diff --git a/dist/sign/index.d.ts b/dist/sign/index.d.ts
index 3f7e08f2f407d3b347729f85c960d17da5a2ac3e..9a601e23836a593e6f38e63958c13663eb6fb6b1 100644
--- a/dist/sign/index.d.ts
+++ b/dist/sign/index.d.ts
@@ -1,6 +1,6 @@
-import { Web3Provider } from '@ethersproject/providers';
-import { Wallet } from '@ethersproject/wallet';
-import { Space, Proposal, CancelProposal, Vote, Follow, Unfollow, Subscribe, Unsubscribe, Profile, Alias } from './types';
+import { Web3Provider } from "@ethersproject/providers";
+import { Wallet } from "@ethersproject/wallet";
+import { Space, Proposal, CancelProposal, Vote, Follow, Unfollow, Subscribe, Unsubscribe, Profile, Alias } from "./types.js";
 export declare const domain: {
     name: string;
     version: string;
diff --git a/dist/utils.d.ts b/dist/utils.d.ts
index 8eace432936661af78bbe1f7126c3bfa85ea4502..2e96c5debeb248b0b3f2e6e1ddd3c95b289113cb 100644
--- a/dist/utils.d.ts
+++ b/dist/utils.d.ts
@@ -1,8 +1,8 @@
-import Multicaller from './utils/multicaller';
-import { getSnapshots } from './utils/blockfinder';
-import getProvider from './utils/provider';
-import { signMessage, getBlockNumber } from './utils/web3';
-import { getHash, verify } from './sign/utils';
+import Multicaller from "./utils/multicaller.js";
+import { getSnapshots } from "./utils/blockfinder.js";
+import getProvider from "./utils/provider.js";
+import { signMessage, getBlockNumber } from "./utils/web3.js";
+import { getHash, verify } from "./sign/utils.js";
 interface Options {
     url?: string;
 }
@@ -28,7 +28,7 @@ export declare function getJSON(uri: any): Promise<any>;
 export declare function ipfsGet(gateway: string, ipfsHash: string, protocolType?: string): Promise<any>;
 export declare function sendTransaction(web3: any, contractAddress: string, abi: any[], action: string, params: any[], overrides?: {}): Promise<any>;
 export declare function getScores(space: string, strategies: Strategy[], network: string, addresses: string[], snapshot?: number | string, scoreApiUrl?: string): Promise<any>;
-export declare function getVp(address: string, network: string, strategies: Strategy[], snapshot: number | 'latest', space: string, delegation: boolean, options?: Options): Promise<any>;
+export declare function getVp(address: string, network: string, strategies: Strategy[], snapshot: number | "latest", space: string, delegation: boolean, options?: Options): Promise<any>;
 export declare function validateSchema(schema: any, data: any): true | import("ajv").ErrorObject<string, Record<string, any>, unknown>[] | null | undefined;
 export declare function getEnsTextRecord(ens: string, record: string, network?: string): Promise<any>;
 export declare function getSpaceUri(id: any, network?: string): Promise<any>;
@@ -54,12 +54,12 @@ declare const _default: {
     sleep: typeof sleep;
     getNumberWithOrdinal: typeof getNumberWithOrdinal;
     voting: {
-        'single-choice': typeof import("./voting/singleChoice").default;
-        approval: typeof import("./voting/approval").default;
-        quadratic: typeof import("./voting/quadratic").default;
-        'ranked-choice': typeof import("./voting/rankedChoice").default;
-        weighted: typeof import("./voting/weighted").default;
-        basic: typeof import("./voting/singleChoice").default;
+        "single-choice": typeof import("./voting/singleChoice.js").default;
+        approval: typeof import("./voting/approval.js").default;
+        quadratic: typeof import("./voting/quadratic.js").default;
+        "ranked-choice": typeof import("./voting/rankedChoice.js").default;
+        weighted: typeof import("./voting/weighted.js").default;
+        basic: typeof import("./voting/singleChoice.js").default;
     };
     getProvider: typeof getProvider;
     signMessage: typeof signMessage;
@@ -67,9 +67,9 @@ declare const _default: {
     Multicaller: typeof Multicaller;
     getSnapshots: typeof getSnapshots;
     validations: {
-        basic: typeof import("./validations/basic").default;
-        aave: typeof import("./validations/aave").default;
-        nouns: typeof import("./validations/nouns").default;
+        basic: typeof import("./validations/basic/index.js").default;
+        aave: typeof import("./validations/aave/index.js").default;
+        nouns: typeof import("./validations/nouns/index.js").default;
     };
     getHash: typeof getHash;
     verify: typeof verify;
diff --git a/dist/validations/index.d.ts b/dist/validations/index.d.ts
index d11e8916699d91f6d5af42e8745f9ed6658dcbd1..e2ffc5d98be770c5e494e65a8d26d633796256d0 100644
--- a/dist/validations/index.d.ts
+++ b/dist/validations/index.d.ts
@@ -1,6 +1,6 @@
-import basic from './basic';
-import aaveSpaceValidation from './aave';
-import nounsSpaceValidation from './nouns';
+import basic from "./basic/index.js";
+import aaveSpaceValidation from "./aave/index.js";
+import nounsSpaceValidation from "./nouns/index.js";
 declare const _default: {
     basic: typeof basic;
     aave: typeof aaveSpaceValidation;
diff --git a/dist/voting/approval.d.ts b/dist/voting/approval.d.ts
index ea6f4d1199907cec0985750015645e3250bd7860..45a90b1ac6d0c0fe9983931c94a2421456953508 100644
--- a/dist/voting/approval.d.ts
+++ b/dist/voting/approval.d.ts
@@ -1,4 +1,4 @@
-import { ApprovalVote, Strategy } from './types';
+import { ApprovalVote, Strategy } from "./types.js";
 export default class ApprovalVoting {
     proposal: {
         choices: string[];
diff --git a/dist/voting/index.d.ts b/dist/voting/index.d.ts
index ab3a8cccce8dd88be0c09162f575e11991fbbd2d..9f752623a3c71134d603329e574a134eebd85f94 100644
--- a/dist/voting/index.d.ts
+++ b/dist/voting/index.d.ts
@@ -1,13 +1,13 @@
-import singleChoice from './singleChoice';
-import approval from './approval';
-import quadratic from './quadratic';
-import rankedChoice from './rankedChoice';
-import weighted from './weighted';
+import singleChoice from "./singleChoice.js";
+import approval from "./approval.js";
+import quadratic from "./quadratic.js";
+import rankedChoice from "./rankedChoice.js";
+import weighted from "./weighted.js";
 declare const _default: {
-    'single-choice': typeof singleChoice;
+    "single-choice": typeof singleChoice;
     approval: typeof approval;
     quadratic: typeof quadratic;
-    'ranked-choice': typeof rankedChoice;
+    "ranked-choice": typeof rankedChoice;
     weighted: typeof weighted;
     basic: typeof singleChoice;
 };
diff --git a/dist/voting/quadratic.d.ts b/dist/voting/quadratic.d.ts
index 690ad4877c3c17cbff6acde5493d9ee1d9d44e2b..ab09a6351412eecfca86185602d33e3d3234a9f8 100644
--- a/dist/voting/quadratic.d.ts
+++ b/dist/voting/quadratic.d.ts
@@ -1,4 +1,4 @@
-import { QuadraticVote, Strategy } from './types';
+import { QuadraticVote, Strategy } from "./types.js";
 export declare function percentageOfTotal(i: any, values: any, total: any): number;
 export declare function quadraticMath(i: any, choice: any, balance: any): number;
 export default class QuadraticVoting {
diff --git a/dist/voting/rankedChoice.d.ts b/dist/voting/rankedChoice.d.ts
index 76eaf43f6f7b1e574cf6e7343c35d8b2ac9ab436..a9fea430c94b47c6d36bdf6cce93abfe8ca201c7 100644
--- a/dist/voting/rankedChoice.d.ts
+++ b/dist/voting/rankedChoice.d.ts
@@ -1,4 +1,4 @@
-import { RankedChoiceVote, Strategy } from './types';
+import { RankedChoiceVote, Strategy } from "./types.js";
 export default class RankedChoiceVoting {
     proposal: {
         choices: string[];
diff --git a/dist/voting/singleChoice.d.ts b/dist/voting/singleChoice.d.ts
index e2ca0a4e574178894ecc40f83d8f3b4b212fedca..538b0ac7ca4a4856b615c051996be5bf281a4a0b 100644
--- a/dist/voting/singleChoice.d.ts
+++ b/dist/voting/singleChoice.d.ts
@@ -1,4 +1,4 @@
-import { SingleChoiceVote, Strategy } from './types';
+import { SingleChoiceVote, Strategy } from "./types.js";
 export default class SingleChoiceVoting {
     proposal: {
         choices: string[];
diff --git a/dist/voting/weighted.d.ts b/dist/voting/weighted.d.ts
index e81d747218fa3130be8119719544aa0177926916..8f23f53626bebd428500681c39e382f2977b1a91 100644
--- a/dist/voting/weighted.d.ts
+++ b/dist/voting/weighted.d.ts
@@ -1,4 +1,4 @@
-import { WeightedVote, Strategy } from './types';
+import { WeightedVote, Strategy } from "./types.js";
 export declare function percentageOfTotal(i: any, values: any, total: any): number;
 export declare function weightedPower(i: any, choice: any, balance: any): number;
 export default class WeightedVoting {
diff --git a/package.json b/package.json
index 5f767229eef2af3cfc74fa1c8b3b7df9a7de22b9..7537e947fdffe098c62d149f1790656549287e51 100644
--- a/package.json
+++ b/package.json
@@ -3,8 +3,8 @@
   "version": "0.4.12",
   "repository": "snapshot-labs/snapshot.js",
   "license": "MIT",
-  "main": "dist/snapshot.cjs.js",
-  "module": "dist/snapshot.esm.js",
+  "type": "module",
+  "main": "dist/snapshot.esm.js",
   "browser": "dist/snapshot.min.js",
   "types": "dist/index.d.ts",
   "dependencies": {