diff --git a/dist/cjs/index.d.ts b/dist/cjs/index.d.ts
index efaf300d1b4efe0ced0492f8179707021d0111b3..642b0aa0537ddf71e14d6ec5e9493ef7d259c0da 100644
--- a/dist/cjs/index.d.ts
+++ b/dist/cjs/index.d.ts
@@ -3,5 +3,5 @@ import { IWalletConnectOptions, IPushServerOptions } from "@walletconnect/types"
 declare class WalletConnect extends Connector {
     constructor(connectorOpts: IWalletConnectOptions, pushServerOpts?: IPushServerOptions);
 }
-export default WalletConnect;
+export = WalletConnect;
 //# sourceMappingURL=index.d.ts.map
diff --git a/dist/esm/index.d.ts b/dist/esm/index.d.ts
index efaf300d1b4efe0ced0492f8179707021d0111b3..642b0aa0537ddf71e14d6ec5e9493ef7d259c0da 100644
--- a/dist/esm/index.d.ts
+++ b/dist/esm/index.d.ts
@@ -3,5 +3,5 @@ import { IWalletConnectOptions, IPushServerOptions } from "@walletconnect/types"
 declare class WalletConnect extends Connector {
     constructor(connectorOpts: IWalletConnectOptions, pushServerOpts?: IPushServerOptions);
 }
-export default WalletConnect;
+export = WalletConnect;
 //# sourceMappingURL=index.d.ts.map