diff --git a/dist/browser-ponyfill.js b/dist/browser-ponyfill.js
index f216aa35a9d32f86df5701c78cf7b039bc2cd47c..7bf27186838bb381fcc705e3839140e7b18410fb 100644
--- a/dist/browser-ponyfill.js
+++ b/dist/browser-ponyfill.js
@@ -1,8 +1,10 @@
 var global = typeof self !== 'undefined' ? self : this;
 var __self__ = (function () {
 function F() {
-this.fetch = false;
-this.DOMException = global.DOMException
+Object.defineProperties(this, {
+  fetch: { value: false, configurable: true, writable: true },
+  DOMException: { value: global.DOMException, configurable: true, writable: true },
+});
 }
 F.prototype = global;
 return new F();
@@ -539,9 +541,11 @@ var irrelevant = (function (exports) {
 
 })({});
 })(__self__);
-__self__.fetch.ponyfill = true;
-// Remove "polyfill" property added by whatwg-fetch
-delete __self__.fetch.polyfill;
+try {
+  __self__.fetch.ponyfill = true;
+  // Remove "polyfill" property added by whatwg-fetch
+  delete __self__.fetch.polyfill;
+} catch (_) {}
 // Choose between native implementation (global) or custom implementation (__self__)
 // var ctx = global.fetch ? global : __self__;
 var ctx = __self__; // this line disable service worker support temporarily