diff --git a/browser.js b/browser.js
index 19b4ba38207e0c60887db2637faea0db7a534e09..945e9102ca20d7b132bec1b4857d753a3342d0e1 100644
--- a/browser.js
+++ b/browser.js
@@ -1,3 +1 @@
-import { Blob } from "blob-polyfill"
-
 export default Blob
diff --git a/index.js b/index.js
index d75efe7b60a1cc18d98266249cccbb4b8d5a54ea..945e9102ca20d7b132bec1b4857d753a3342d0e1 100644
--- a/index.js
+++ b/index.js
@@ -1,3 +1 @@
-import { Blob } from "fetch-blob"
-
 export default Blob