diff --git a/dist/cjs/fortmatic.js b/dist/cjs/fortmatic.cjs
similarity index 100%
rename from dist/cjs/fortmatic.js
rename to dist/cjs/fortmatic.cjs
diff --git a/dist/cjs/src/core/fm-payload-transport.d.ts b/dist/cjs/src/core/fm-payload-transport.d.ts
index 5a8ddb844fc45f42570980e88089c1bf2ebf6691..73a9cdb2252eae4afd27fd566761b857ef55adc1 100644
--- a/dist/cjs/src/core/fm-payload-transport.d.ts
+++ b/dist/cjs/src/core/fm-payload-transport.d.ts
@@ -1,5 +1,5 @@
-import { FmIncomingWindowMessage, FmMessageEvent, FmOutgoingWindowMessage, JsonRpcBatchRequestPayload, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types';
-import { FmIframeController } from './fm-iframe-controller';
+import { FmIncomingWindowMessage, FmMessageEvent, FmOutgoingWindowMessage, JsonRpcBatchRequestPayload, JsonRpcRequestPayload, JsonRpcResponsePayload } from "../types/index.js";
+import { FmIframeController } from "./fm-iframe-controller.js";
 interface RemoveEventListenerFunction {
     /**
      * Stop listening on the event associated with this `FmFetchOffFunction`
diff --git a/dist/cjs/src/core/fm-provider.d.ts b/dist/cjs/src/core/fm-provider.d.ts
index 8b7a36e74a59aaca48b2bb4f45b408bf4ea5fb48..515b5c74e45b37cf430d88a98892d714fa9efcf2 100644
--- a/dist/cjs/src/core/fm-provider.d.ts
+++ b/dist/cjs/src/core/fm-provider.d.ts
@@ -1,4 +1,4 @@
-import { JsonRpcBatchRequestPayload, JsonRpcRequestCallback, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types';
+import { JsonRpcBatchRequestPayload, JsonRpcRequestCallback, JsonRpcRequestPayload, JsonRpcResponsePayload } from "../types/index.js";
 /**
  * Fortmatic's Web3-compliant provider.
  */
diff --git a/dist/cjs/src/core/json-rpc-response.d.ts b/dist/cjs/src/core/json-rpc-response.d.ts
index 9c0cac28c2849353e0a0291037d0c8f65be1305d..e210b26bdb3337be41eeecce90fde9463a661c69 100644
--- a/dist/cjs/src/core/json-rpc-response.d.ts
+++ b/dist/cjs/src/core/json-rpc-response.d.ts
@@ -1,4 +1,4 @@
-import { JsonRpcBatchRequestPayload, JsonRpcError, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types';
+import { JsonRpcBatchRequestPayload, JsonRpcError, JsonRpcRequestPayload, JsonRpcResponsePayload } from "../types/index.js";
 /**
  * A class which standardizes JSON RPC responses to ensure the correct
  * formatting.
diff --git a/dist/cjs/src/core/sdk-exceptions.d.ts b/dist/cjs/src/core/sdk-exceptions.d.ts
index bf6f4c611fbb2c1c14d6a0695a169f255dac673e..a881069a2f0e6f3dc93dd0fc46530685fd41c712 100644
--- a/dist/cjs/src/core/sdk-exceptions.d.ts
+++ b/dist/cjs/src/core/sdk-exceptions.d.ts
@@ -1,4 +1,4 @@
-import { SDKErrorCode, SDKWarningCode, JsonRpcError, RPCErrorCode } from '../types';
+import { SDKErrorCode, SDKWarningCode, JsonRpcError, RPCErrorCode } from "../types/index.js";
 export declare class FortmaticError extends Error {
     code: SDKErrorCode;
     __proto__: ErrorConstructor;
diff --git a/dist/cjs/src/core/sdk.d.ts b/dist/cjs/src/core/sdk.d.ts
index 311d81decb72319d4123b9e9bb7d778f72494e39..9ef9a77e8a1adf4c12489e6ab4413c07ebfc6c3e 100644
--- a/dist/cjs/src/core/sdk.d.ts
+++ b/dist/cjs/src/core/sdk.d.ts
@@ -1,9 +1,9 @@
-import { PhantomUser } from '../modules/phantom-mode/phantom-user';
-import { TransactionsModule } from '../modules/widget-mode/transactions-module';
-import { UserModule } from '../modules/widget-mode/user-module';
-import { LoginWithMagicLinkConfiguration, WidgetModeConfiguration, WidgetModeGSNConfiguration } from '../types';
-import { FmProvider } from './fm-provider';
-import { QueryParameters } from '../util/query-params';
+import { PhantomUser } from "../modules/phantom-mode/phantom-user.js";
+import { TransactionsModule } from "../modules/widget-mode/transactions-module.js";
+import { UserModule } from "../modules/widget-mode/user-module.js";
+import { LoginWithMagicLinkConfiguration, WidgetModeConfiguration, WidgetModeGSNConfiguration } from "../types/index.js";
+import { FmProvider } from "./fm-provider.js";
+import { QueryParameters } from "../util/query-params.js";
 /**
  * Base class for Fortmatic SDKs
  */
@@ -15,7 +15,7 @@ export declare abstract class SDK {
     readonly gsnRelay?: boolean;
     constructor(options: {
         apiKey: string;
-        ethNetwork?: QueryParameters['ETH_NETWORK'];
+        ethNetwork?: QueryParameters["ETH_NETWORK"];
         endpoint: string;
         gsnRelay?: boolean;
     });
@@ -29,7 +29,7 @@ export declare abstract class SDK {
  */
 export declare class PhantomMode extends SDK {
     readonly user: PhantomUser;
-    constructor(apiKey: string, ethNetwork?: QueryParameters['ETH_NETWORK']);
+    constructor(apiKey: string, ethNetwork?: QueryParameters["ETH_NETWORK"]);
     /**
      * Initiate the "magic link" login flow for a user.
      */
@@ -42,7 +42,7 @@ export declare class WidgetMode extends SDK {
     static readonly Phantom: typeof PhantomMode;
     readonly transactions: TransactionsModule;
     readonly user: UserModule;
-    constructor(apiKey: string, ethNetwork?: QueryParameters['ETH_NETWORK'], gsnRelayOptions?: WidgetModeGSNConfiguration);
+    constructor(apiKey: string, ethNetwork?: QueryParameters["ETH_NETWORK"], gsnRelayOptions?: WidgetModeGSNConfiguration);
     /**
      * Apply configuration to customize Fortmatic's Ethereum functionality.
      */
diff --git a/dist/cjs/src/index.d.ts b/dist/cjs/src/index.d.ts
index 7f30790a2e3fd4563c44b66960fac0f138dd759d..aed6f7b55d5f9edda7a0e032885c8c3a56f2ed53 100644
--- a/dist/cjs/src/index.d.ts
+++ b/dist/cjs/src/index.d.ts
@@ -1,5 +1,5 @@
-import { Fortmatic } from './core/sdk';
-import { FortmaticError, FortmaticWarning, RpcError } from './core/sdk-exceptions';
+import { Fortmatic } from "./core/sdk.js";
+import { FortmaticError, FortmaticWarning, RpcError } from "./core/sdk-exceptions.js";
 export { Fortmatic as default };
 export { FortmaticError, FortmaticWarning, RpcError };
-export * from './types';
+export * from "./types/index.js";
diff --git a/dist/cjs/src/modules/base-module.d.ts b/dist/cjs/src/modules/base-module.d.ts
index 0728cd576f7cc4681cf5aa03c48b764bfcd3327d..cddd35a3b279b65f090b736e38e78b9b7ba1d759 100644
--- a/dist/cjs/src/modules/base-module.d.ts
+++ b/dist/cjs/src/modules/base-module.d.ts
@@ -1,4 +1,4 @@
-import { SDK } from '../core/sdk';
+import { SDK } from "../core/sdk.js";
 export declare class BaseModule {
     protected readonly sdk: SDK;
     constructor(sdk: SDK);
diff --git a/dist/cjs/src/modules/phantom-mode/phantom-user.d.ts b/dist/cjs/src/modules/phantom-mode/phantom-user.d.ts
index 71b092ec620911aa5ee9b38e8341d2a9806fb401..25b0fb49abfbdedeedf6c5c10d262f9e306624db 100644
--- a/dist/cjs/src/modules/phantom-mode/phantom-user.d.ts
+++ b/dist/cjs/src/modules/phantom-mode/phantom-user.d.ts
@@ -1,5 +1,5 @@
-import { GetIdTokenConfiguration, PhantomModeUserMetadata } from '../../types';
-import { BaseModule } from '../base-module';
+import { GetIdTokenConfiguration, PhantomModeUserMetadata } from "../../types/index.js";
+import { BaseModule } from "../base-module.js";
 /**
  * A stateless object representing the current Fortmatic Auth user.
  */
diff --git a/dist/cjs/src/modules/widget-mode/transactions-module.d.ts b/dist/cjs/src/modules/widget-mode/transactions-module.d.ts
index dcbf535b9081f39fd2464d5c0165083534b804a7..4d19ee96aa54e5b5c1a65a021342a98b7fbf498c 100644
--- a/dist/cjs/src/modules/widget-mode/transactions-module.d.ts
+++ b/dist/cjs/src/modules/widget-mode/transactions-module.d.ts
@@ -1,5 +1,5 @@
-import { ComposeTransactionConfig } from '../../types';
-import { BaseModule } from '../base-module';
+import { ComposeTransactionConfig } from "../../types/index.js";
+import { BaseModule } from "../base-module.js";
 /**
  *
  */
diff --git a/dist/cjs/src/modules/widget-mode/user-module.d.ts b/dist/cjs/src/modules/widget-mode/user-module.d.ts
index 2ae620d70ee74b5261e55b2cb568ae80d7551d4b..f697001a00efbf4a63a72b36de81f2fc87c7e801 100644
--- a/dist/cjs/src/modules/widget-mode/user-module.d.ts
+++ b/dist/cjs/src/modules/widget-mode/user-module.d.ts
@@ -1,5 +1,5 @@
-import { DepositConfiguration } from '../../types';
-import { BaseModule } from '../base-module';
+import { DepositConfiguration } from "../../types/index.js";
+import { BaseModule } from "../base-module.js";
 /**
  *
  */
diff --git a/dist/cjs/src/types/core/index.d.ts b/dist/cjs/src/types/core/index.d.ts
index ea8aecd1420ffc953a544a9caaf67dc9fa69fecc..a76bec54f316d9ad71891f9bed2e93cf7e707044 100644
--- a/dist/cjs/src/types/core/index.d.ts
+++ b/dist/cjs/src/types/core/index.d.ts
@@ -1,4 +1,4 @@
-export * from './json-rpc-types';
-export * from './provider-types';
-export * from './fm-message-types';
-export * from './exception-types';
+export * from "./json-rpc-types.js";
+export * from "./provider-types.js";
+export * from "./fm-message-types.js";
+export * from "./exception-types.js";
diff --git a/dist/cjs/src/types/core/provider-types.d.ts b/dist/cjs/src/types/core/provider-types.d.ts
index f351288980508833db711288270fa1e37261ffb0..80829410fdd2e1839db699c9ba322b44570a738c 100644
--- a/dist/cjs/src/types/core/provider-types.d.ts
+++ b/dist/cjs/src/types/core/provider-types.d.ts
@@ -1,4 +1,4 @@
-import { JsonRpcBatchRequestPayload, JsonRpcError, JsonRpcRequestCallback, JsonRpcRequestPayload, JsonRpcResponsePayload } from './json-rpc-types';
+import { JsonRpcBatchRequestPayload, JsonRpcError, JsonRpcRequestCallback, JsonRpcRequestPayload, JsonRpcResponsePayload } from "./json-rpc-types.js";
 /**
  * The shape of payloads encoded by `FmProvider` while they are queued.
  */
diff --git a/dist/cjs/src/types/index.d.ts b/dist/cjs/src/types/index.d.ts
index 1a409b557e613e63234630feb52dcd1b9eaace04..bec153219411a3971247091002931d969bc61775 100644
--- a/dist/cjs/src/types/index.d.ts
+++ b/dist/cjs/src/types/index.d.ts
@@ -1,3 +1,3 @@
-export * from './core';
-export * from './widget-mode';
-export * from './phantom-mode';
+export * from "./core/index.js";
+export * from "./widget-mode.js";
+export * from "./phantom-mode.js";
diff --git a/dist/cjs/src/util/emit-payload-promise.d.ts b/dist/cjs/src/util/emit-payload-promise.d.ts
index 8dc89ad4a40c79d4c4d390b7c361b3c2a9fb5ea8..076143013f50c8719af4b4c22dc24a9eef28d2da 100644
--- a/dist/cjs/src/util/emit-payload-promise.d.ts
+++ b/dist/cjs/src/util/emit-payload-promise.d.ts
@@ -1,5 +1,5 @@
-import { FmProvider } from '../core/fm-provider';
-import { JsonRpcRequestPayload } from '../types';
+import { FmProvider } from "../core/fm-provider.js";
+import { JsonRpcRequestPayload } from "../types/index.js";
 /**
  * Emit a payload to the provider asynchronously using native JavaScript
  * Promises.
diff --git a/dist/cjs/src/util/json-rpc-helpers.d.ts b/dist/cjs/src/util/json-rpc-helpers.d.ts
index f0ed19c7065db7c5e65b0f22a88bbf36059ff278..ef73b9ffa7ace97dc603f3e940d9efab568c7377 100644
--- a/dist/cjs/src/util/json-rpc-helpers.d.ts
+++ b/dist/cjs/src/util/json-rpc-helpers.d.ts
@@ -1,4 +1,4 @@
-import { ComposeTransactionConfig, JsonRpcBatchRequestPayload, JsonRpcRequestPayload } from '../types';
+import { ComposeTransactionConfig, JsonRpcBatchRequestPayload, JsonRpcRequestPayload } from "../types/index.js";
 export declare const JSON_RPC_VERSION = "2.0";
 /**
  * Build a valid JSON RPC payload for emitting to the Ethereum node or Fortmatic
diff --git a/dist/cjs/src/util/type-guards.d.ts b/dist/cjs/src/util/type-guards.d.ts
index 9ee6ac9f9dd8b792f9637157757484a83a3e2e65..c45fb6d761098668aa466a2ace0258a5a2310dc0 100644
--- a/dist/cjs/src/util/type-guards.d.ts
+++ b/dist/cjs/src/util/type-guards.d.ts
@@ -7,7 +7,7 @@
  * @see
  * https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types
  */
-import { FmBatchRequest, FmPayloadMethod, FmRequest, JsonRpcBatchRequestPayload, JsonRpcRequestPayload, JsonRpcResponsePayload, RPCErrorCode } from '../types';
+import { FmBatchRequest, FmPayloadMethod, FmRequest, JsonRpcBatchRequestPayload, JsonRpcRequestPayload, JsonRpcResponsePayload, RPCErrorCode } from "../types/index.js";
 /**
  * Assert `value` is a `JsonRpcBatchRequestPayload` object.
  */
diff --git a/dist/fortmatic.js b/dist/fortmatic.cjs
similarity index 100%
rename from dist/fortmatic.js
rename to dist/fortmatic.cjs
diff --git a/package.json b/package.json
index f246a9c50d46f47d8c675b0eb88868d49f1beb89..c1c38e2db89469071222b3639fd9b07ddd282111 100644
--- a/package.json
+++ b/package.json
@@ -20,7 +20,8 @@
     "dapp"
   ],
   "homepage": "https://www.fortmatic.com",
-  "main": "dist/cjs/fortmatic.js",
+  "type": "module",
+  "main": "dist/cjs/fortmatic.cjs",
   "types": "dist/cjs/src/index.d.ts",
   "scripts": {
     "start": "npm run clean:build && ./scripts/start.sh",