diff --git a/dist/cjs/src/core/fm-payload-transport.d.ts b/dist/cjs/src/core/fm-payload-transport.d.ts
index 5a8ddb844fc45f42570980e88089c1bf2ebf6691..ed9eb4fe13faeaeef82e9d8941533ff8b69d9fcb 100644
--- a/dist/cjs/src/core/fm-payload-transport.d.ts
+++ b/dist/cjs/src/core/fm-payload-transport.d.ts
@@ -1,5 +1,5 @@
-import { FmIncomingWindowMessage, FmMessageEvent, FmOutgoingWindowMessage, JsonRpcBatchRequestPayload, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types';
-import { FmIframeController } from './fm-iframe-controller';
+import { FmIncomingWindowMessage, FmMessageEvent, FmOutgoingWindowMessage, JsonRpcBatchRequestPayload, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types/index.js';
+import { FmIframeController } from './fm-iframe-controller.js';
 interface RemoveEventListenerFunction {
     /**
      * Stop listening on the event associated with this `FmFetchOffFunction`
diff --git a/dist/cjs/src/core/fm-provider.d.ts b/dist/cjs/src/core/fm-provider.d.ts
index 3fae1acc32ca87baea50d24794cb1f6ff57aedc7..f9d834b0e6e23e31d458635540ac1cc64c4eb89d 100644
--- a/dist/cjs/src/core/fm-provider.d.ts
+++ b/dist/cjs/src/core/fm-provider.d.ts
@@ -1,4 +1,4 @@
-import { JsonRpcBatchRequestPayload, JsonRpcRequestCallback, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types';
+import { JsonRpcBatchRequestPayload, JsonRpcRequestCallback, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types/index.js';
 /**
  * Fortmatic's Web3-compliant provider.
  */
diff --git a/dist/cjs/src/core/json-rpc-response.d.ts b/dist/cjs/src/core/json-rpc-response.d.ts
index 9c0cac28c2849353e0a0291037d0c8f65be1305d..e5fd4a163c62b21327b9b4bc3452f9566d0c8ba9 100644
--- a/dist/cjs/src/core/json-rpc-response.d.ts
+++ b/dist/cjs/src/core/json-rpc-response.d.ts
@@ -1,4 +1,4 @@
-import { JsonRpcBatchRequestPayload, JsonRpcError, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types';
+import { JsonRpcBatchRequestPayload, JsonRpcError, JsonRpcRequestPayload, JsonRpcResponsePayload } from '../types/index.js';
 /**
  * A class which standardizes JSON RPC responses to ensure the correct
  * formatting.
diff --git a/dist/cjs/src/core/sdk-exceptions.d.ts b/dist/cjs/src/core/sdk-exceptions.d.ts
index bf6f4c611fbb2c1c14d6a0695a169f255dac673e..c4fae5cbb3607155ca599b41512207fa0f1c0829 100644
--- a/dist/cjs/src/core/sdk-exceptions.d.ts
+++ b/dist/cjs/src/core/sdk-exceptions.d.ts
@@ -1,4 +1,4 @@
-import { SDKErrorCode, SDKWarningCode, JsonRpcError, RPCErrorCode } from '../types';
+import { SDKErrorCode, SDKWarningCode, JsonRpcError, RPCErrorCode } from '../types/index.js';
 export declare class FortmaticError extends Error {
     code: SDKErrorCode;
     __proto__: ErrorConstructor;
diff --git a/dist/cjs/src/core/sdk.d.ts b/dist/cjs/src/core/sdk.d.ts
index 311d81decb72319d4123b9e9bb7d778f72494e39..faccfb212db6f5de541276fde8f7950528cac4c0 100644
--- a/dist/cjs/src/core/sdk.d.ts
+++ b/dist/cjs/src/core/sdk.d.ts
@@ -1,9 +1,9 @@
-import { PhantomUser } from '../modules/phantom-mode/phantom-user';
-import { TransactionsModule } from '../modules/widget-mode/transactions-module';
-import { UserModule } from '../modules/widget-mode/user-module';
-import { LoginWithMagicLinkConfiguration, WidgetModeConfiguration, WidgetModeGSNConfiguration } from '../types';
-import { FmProvider } from './fm-provider';
-import { QueryParameters } from '../util/query-params';
+import { PhantomUser } from '../modules/phantom-mode/phantom-user.js';
+import { TransactionsModule } from '../modules/widget-mode/transactions-module.js';
+import { UserModule } from '../modules/widget-mode/user-module.js';
+import { LoginWithMagicLinkConfiguration, WidgetModeConfiguration, WidgetModeGSNConfiguration } from '../types/index.js';
+import { FmProvider } from './fm-provider.js';
+import { QueryParameters } from '../util/query-params.js';
 /**
  * Base class for Fortmatic SDKs
  */
diff --git a/dist/cjs/src/index.d.ts b/dist/cjs/src/index.d.ts
index 7f30790a2e3fd4563c44b66960fac0f138dd759d..8270fd03ccc991007f3119b090e2ff34b99eca83 100644
--- a/dist/cjs/src/index.d.ts
+++ b/dist/cjs/src/index.d.ts
@@ -1,5 +1,5 @@
-import { Fortmatic } from './core/sdk';
-import { FortmaticError, FortmaticWarning, RpcError } from './core/sdk-exceptions';
+import { Fortmatic } from './core/sdk.js';
+import { FortmaticError, FortmaticWarning, RpcError } from './core/sdk-exceptions.js';
 export { Fortmatic as default };
 export { FortmaticError, FortmaticWarning, RpcError };
-export * from './types';
+export * from './types/index.js';
diff --git a/dist/cjs/src/modules/base-module.d.ts b/dist/cjs/src/modules/base-module.d.ts
index 0728cd576f7cc4681cf5aa03c48b764bfcd3327d..ce22aab02d2b75924fc8d8f6c62b618564c0626d 100644
--- a/dist/cjs/src/modules/base-module.d.ts
+++ b/dist/cjs/src/modules/base-module.d.ts
@@ -1,4 +1,4 @@
-import { SDK } from '../core/sdk';
+import { SDK } from '../core/sdk.js';
 export declare class BaseModule {
     protected readonly sdk: SDK;
     constructor(sdk: SDK);
diff --git a/dist/cjs/src/modules/phantom-mode/phantom-user.d.ts b/dist/cjs/src/modules/phantom-mode/phantom-user.d.ts
index 71b092ec620911aa5ee9b38e8341d2a9806fb401..d3ff2af290a501e44b7f0fa8b9e0414f84dbd918 100644
--- a/dist/cjs/src/modules/phantom-mode/phantom-user.d.ts
+++ b/dist/cjs/src/modules/phantom-mode/phantom-user.d.ts
@@ -1,5 +1,5 @@
-import { GetIdTokenConfiguration, PhantomModeUserMetadata } from '../../types';
-import { BaseModule } from '../base-module';
+import { GetIdTokenConfiguration, PhantomModeUserMetadata } from '../../types/index.js';
+import { BaseModule } from '../base-module.js';
 /**
  * A stateless object representing the current Fortmatic Auth user.
  */
diff --git a/dist/cjs/src/modules/widget-mode/transactions-module.d.ts b/dist/cjs/src/modules/widget-mode/transactions-module.d.ts
index dcbf535b9081f39fd2464d5c0165083534b804a7..2427ce4dbbfc7939fc20b1913242f016a043aa48 100644
--- a/dist/cjs/src/modules/widget-mode/transactions-module.d.ts
+++ b/dist/cjs/src/modules/widget-mode/transactions-module.d.ts
@@ -1,5 +1,5 @@
-import { ComposeTransactionConfig } from '../../types';
-import { BaseModule } from '../base-module';
+import { ComposeTransactionConfig } from '../../types/index.js';
+import { BaseModule } from '../base-module.js';
 /**
  *
  */
diff --git a/dist/cjs/src/modules/widget-mode/user-module.d.ts b/dist/cjs/src/modules/widget-mode/user-module.d.ts
index 2ae620d70ee74b5261e55b2cb568ae80d7551d4b..694ec5c001c51be3b02673df557a51f693885600 100644
--- a/dist/cjs/src/modules/widget-mode/user-module.d.ts
+++ b/dist/cjs/src/modules/widget-mode/user-module.d.ts
@@ -1,5 +1,5 @@
-import { DepositConfiguration } from '../../types';
-import { BaseModule } from '../base-module';
+import { DepositConfiguration } from '../../types/index.js';
+import { BaseModule } from '../base-module.js';
 /**
  *
  */
diff --git a/dist/cjs/src/types/core/index.d.ts b/dist/cjs/src/types/core/index.d.ts
index ea8aecd1420ffc953a544a9caaf67dc9fa69fecc..e280dd97b136c052d2e9da01de6cb86d51f901ad 100644
--- a/dist/cjs/src/types/core/index.d.ts
+++ b/dist/cjs/src/types/core/index.d.ts
@@ -1,4 +1,4 @@
-export * from './json-rpc-types';
-export * from './provider-types';
-export * from './fm-message-types';
-export * from './exception-types';
+export * from './json-rpc-types.js';
+export * from './provider-types.js';
+export * from './fm-message-types.js';
+export * from './exception-types.js';
diff --git a/dist/cjs/src/types/core/provider-types.d.ts b/dist/cjs/src/types/core/provider-types.d.ts
index f351288980508833db711288270fa1e37261ffb0..51c9f2060c8bb5810837ed73fc71098de468a8ae 100644
--- a/dist/cjs/src/types/core/provider-types.d.ts
+++ b/dist/cjs/src/types/core/provider-types.d.ts
@@ -1,4 +1,4 @@
-import { JsonRpcBatchRequestPayload, JsonRpcError, JsonRpcRequestCallback, JsonRpcRequestPayload, JsonRpcResponsePayload } from './json-rpc-types';
+import { JsonRpcBatchRequestPayload, JsonRpcError, JsonRpcRequestCallback, JsonRpcRequestPayload, JsonRpcResponsePayload } from './json-rpc-types.js';
 /**
  * The shape of payloads encoded by `FmProvider` while they are queued.
  */
diff --git a/dist/cjs/src/types/index.d.ts b/dist/cjs/src/types/index.d.ts
index 1a409b557e613e63234630feb52dcd1b9eaace04..ec2e991aeff1385bd05a754d5fdada77587961d5 100644
--- a/dist/cjs/src/types/index.d.ts
+++ b/dist/cjs/src/types/index.d.ts
@@ -1,3 +1,3 @@
-export * from './core';
-export * from './widget-mode';
-export * from './phantom-mode';
+export * from './core/index.js';
+export * from './widget-mode.js';
+export * from './phantom-mode.js';
diff --git a/dist/cjs/src/util/emit-payload-promise.d.ts b/dist/cjs/src/util/emit-payload-promise.d.ts
index 8dc89ad4a40c79d4c4d390b7c361b3c2a9fb5ea8..350b7a0dda32e9aeb9ac0c3dff89f27bffc8fe9c 100644
--- a/dist/cjs/src/util/emit-payload-promise.d.ts
+++ b/dist/cjs/src/util/emit-payload-promise.d.ts
@@ -1,5 +1,5 @@
-import { FmProvider } from '../core/fm-provider';
-import { JsonRpcRequestPayload } from '../types';
+import { FmProvider } from '../core/fm-provider.js';
+import { JsonRpcRequestPayload } from '../types/index.js';
 /**
  * Emit a payload to the provider asynchronously using native JavaScript
  * Promises.
diff --git a/dist/cjs/src/util/json-rpc-helpers.d.ts b/dist/cjs/src/util/json-rpc-helpers.d.ts
index f0ed19c7065db7c5e65b0f22a88bbf36059ff278..c34800039768167736f3d45b96e9272ea9f556b8 100644
--- a/dist/cjs/src/util/json-rpc-helpers.d.ts
+++ b/dist/cjs/src/util/json-rpc-helpers.d.ts
@@ -1,4 +1,4 @@
-import { ComposeTransactionConfig, JsonRpcBatchRequestPayload, JsonRpcRequestPayload } from '../types';
+import { ComposeTransactionConfig, JsonRpcBatchRequestPayload, JsonRpcRequestPayload } from '../types/index.js';
 export declare const JSON_RPC_VERSION = "2.0";
 /**
  * Build a valid JSON RPC payload for emitting to the Ethereum node or Fortmatic
diff --git a/dist/fortmatic.js b/dist/fortmatic.cjs
similarity index 100%
rename from dist/fortmatic.js
rename to dist/fortmatic.cjs
diff --git a/package.json b/package.json
index 70fc2b9252138b2bc36447313d5ed28d2ec2e110..c56f462bbd386a24b85a3a71bc7d042cb44fe211 100644
--- a/package.json
+++ b/package.json
@@ -2,6 +2,8 @@
   "name": "fortmatic",
   "version": "2.4.0",
   "description": "Fortmatic Javascript SDK",
+  "type": "module",
+  "main": "dist/cjs/fortmatic.cjs",
   "author": "Fortmatic <team@fortmatic.com> (https://fortmatic.com/)",
   "license": "MIT",
   "repository": {
@@ -20,7 +22,6 @@
     "dapp"
   ],
   "homepage": "https://www.fortmatic.com",
-  "main": "dist/cjs/fortmatic.js",
   "types": "dist/cjs/src/index.d.ts",
   "scripts": {
     "start": "npm run clean:build && ./scripts/start.sh",
