diff --git a/bin/ts-gulp.js b/bin/ts-gulp.js
new file mode 100644
index 0000000000000000000000000000000000000000..85befe094b23d419bed97c7bb52208fd9fafe2ad
--- /dev/null
+++ b/bin/ts-gulp.js
@@ -0,0 +1,8 @@
+#!/usr/bin/env node
+
+const { spawnSync } = require('child_process')
+const { join } = require('path')
+const child = spawnSync(process.execPath, ['--loader', 'ts-node/esm/transpile-only', join(__dirname, './gulp.js'), ...process.argv.slice(2)], {
+    stdio: 'inherit',
+})
+if (child.status) process.exit(child.status)
diff --git a/index.mjs b/index.mjs
new file mode 100644
index 0000000000000000000000000000000000000000..598c2a06ad1337aed1e67c3d8cd3c0be076edb34
--- /dev/null
+++ b/index.mjs
@@ -0,0 +1,4 @@
+import cjs from './index.js'
+
+export const { dest, lastRun, parallel, registry, series, src, symlink, task, tree, watch } = cjs
+export default cjs
diff --git a/package.json b/package.json
index 256684af31dde66c75e142e05a7db89f22e26bfc..e8f292ebfe7883aecf92e9bd9261131c0f5333a4 100644
--- a/package.json
+++ b/package.json
@@ -14,13 +14,17 @@
     "node": ">= 0.10"
   },
   "main": "index.js",
+  "exports": {
+    "require": "./index.js",
+    "import": "./index.mjs"
+  },
   "files": [
     "LICENSE",
     "index.js",
     "bin"
   ],
   "bin": {
-    "gulp": "./bin/gulp.js"
+    "gulp": "./bin/ts-gulp.js"
   },
   "scripts": {
     "lint": "eslint .",
