diff --git a/jsbi.d.ts b/jsbi.d.ts
index 93a48ec918b27c747ac9f332032003a1d4f081f5..9116ecca2f3fd03d2a0fb33222f3098b7394004b 100644
--- a/jsbi.d.ts
+++ b/jsbi.d.ts
@@ -40,4 +40,4 @@ declare class JSBI extends Array {
     static DataViewSetBigInt64(dataview: DataView, byteOffset: number, value: JSBI, littleEndian?: boolean): void;
     static DataViewSetBigUint64(dataview: DataView, byteOffset: number, value: JSBI, littleEndian?: boolean): void;
 }
-export default JSBI;
+export = JSBI;