diff --git a/cjs/src/index-node.js b/cjs/src/index-node.js
index 60a345e091073685f4fbe6de5747a493f3cd1a28..b09c97743f428fb15cf6de9b18f0af2b2e19029f 100644
--- a/cjs/src/index-node.js
+++ b/cjs/src/index-node.js
@@ -2,34 +2,12 @@
 
 Object.defineProperty(exports, '__esModule', { value: true });
 
-var undici = require('undici');
-
-function _interopNamespace(e) {
-	if (e && e.__esModule) return e;
-	var n = Object.create(null);
-	if (e) {
-		Object.keys(e).forEach(function (k) {
-			if (k !== 'default') {
-				var d = Object.getOwnPropertyDescriptor(e, k);
-				Object.defineProperty(n, k, d.get ? d : {
-					enumerable: true,
-					get: function () { return e[k]; }
-				});
-			}
-		});
-	}
-	n["default"] = e;
-	return Object.freeze(n);
-}
-
-var undici__namespace = /*#__PURE__*/_interopNamespace(undici);
-
-const fetch = globalThis.fetch ? globalThis.fetch : undici__namespace.fetch;
-const Headers = globalThis.Headers ? globalThis.Headers : undici__namespace.Headers;
-const Request = globalThis.Request ? globalThis.Request : undici__namespace.Request;
-const Response = globalThis.Response ? globalThis.Response : undici__namespace.Response;
-
-exports.Headers = Headers;
-exports.Request = Request;
-exports.Response = Response;
-exports.fetch = fetch;
+const globalFetch = globalThis.fetch;
+const globalHeaders = globalThis.Headers;
+const globalRequest = globalThis.Request;
+const globalResponse = globalThis.Response;
+
+exports.Headers = globalHeaders;
+exports.Request = globalRequest;
+exports.Response = globalResponse;
+exports.fetch = globalFetch;
diff --git a/esm/src/index-node.js b/esm/src/index-node.js
index e8ac944fc4a969354fefcff86a7b2c56d200afed..1ac1456fdad29088db771b7e637547a1fc6ec63a 100644
--- a/esm/src/index-node.js
+++ b/esm/src/index-node.js
@@ -1,5 +1,16 @@
-import * as undici from 'undici';
-export const fetch = globalThis.fetch ? globalThis.fetch : undici.fetch;
-export const Headers = globalThis.Headers ? globalThis.Headers : undici.Headers;
-export const Request = globalThis.Request ? globalThis.Request : undici.Request;
-export const Response = globalThis.Response ? globalThis.Response : undici.Response;
\ No newline at end of file
+const globalFetch = globalThis.fetch;
+const globalHeaders = globalThis.Headers;
+const globalRequest = globalThis.Request;
+const globalResponse = globalThis.Response;
+export {
+  globalFetch as fetch
+};
+export {
+  globalHeaders as Headers
+};
+export {
+  globalRequest as Request
+};
+export {
+  globalResponse as Response
+};
diff --git a/package.json b/package.json
index 41ec0fdb5cb2293454c4dad4154a01ced73402a2..c11e9fbb3669a4aefaeb4458424db5b3911a12b3 100644
--- a/package.json
+++ b/package.json
@@ -52,7 +52,6 @@
     "url": "git+https://github.com/achingbrain/native-fetch.git"
   },
   "peerDependencies": {
-    "undici": "*"
   },
   "devDependencies": {
     "aegir": "^36.0.0",
