diff --git a/dist/index.js b/dist/index.js
index 292acdeb2741227691616e3571ff7fc8fa0a19fc..9ed88493b1bc767a5a2fcae45f521eb66979b467 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -170,7 +170,7 @@ var QRCode = /** @class */ (function (_super) {
         var qrCode = qrGenerator(0, ecLevel);
         qrCode.addData(QRCode.utf16to8(value));
         qrCode.make();
-        var canvas = ReactDOM.findDOMNode(this.canvas.current);
+        var canvas = (this.canvas.current);
         var ctx = canvas.getContext('2d');
         var canvasSize = size + (2 * quietZone);
         var length = qrCode.getModuleCount();
