diff --git a/dist/backend.js b/dist/backend.js
index 0c65c8dc4a034f724c045fcc433983a8aa8f4843..e0ea6b50420e8ed3703c774849f522a733d696a6 100644
--- a/dist/backend.js
+++ b/dist/backend.js
@@ -12406,7 +12406,7 @@ function attach(hook, rendererID, renderer, global) {
     unpatchConsoleForStrictMode: unpatchForStrictMode,
     updateComponentFilters
   };
-}
+}if (sessionStorageGetItem("React::DevTools::reloadAndProfile") === "true") globalThis.__REACT_DEVTOOLS_ATTACH__ = attach
 ;// CONCATENATED MODULE: ../react-devtools-shared/src/backend/console.js
 /**
  * Copyright (c) Meta Platforms, Inc. and affiliates.
