diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 51ec13dfbed4b67c9c8a245b7f4774f989422658..0000000000000000000000000000000000000000
diff --git a/esm/useAsync.js b/esm/useAsync.js
index ddb0bb0370fc30090cf9a0ecc74bc039800a89ed..901d7bf1aa5c794d3bc12c41aaea2208052271b5 100644
--- a/esm/useAsync.js
+++ b/esm/useAsync.js
@@ -6,7 +6,11 @@ export default function useAsync(fn, deps) {
         loading: true,
     }), state = _a[0], callback = _a[1];
     useEffect(function () {
-        callback();
+        try {
+            callback();
+        } catch (error) {
+            // do nothing
+        }
     }, [callback]);
     return state;
 }
diff --git a/esm/useAsyncFn.js b/esm/useAsyncFn.js
index 01d7307bd106229f6d791d0c588589235f404d97..599b7371c29ea53dfd9c831f5db61eb9dfd60f13 100644
--- a/esm/useAsyncFn.js
+++ b/esm/useAsyncFn.js
@@ -13,15 +13,16 @@ export default function useAsyncFn(fn, deps, initialState) {
             args[_i] = arguments[_i];
         }
         var callId = ++lastCallId.current;
-        if (!state.loading) {
-            set(function (prevState) { return (__assign(__assign({}, prevState), { loading: true })); });
-        }
+        set(function (prevState) {
+            if (prevState.loading) return prevState
+            return (__assign(__assign({}, prevState), { loading: true })); 
+        });
         return fn.apply(void 0, args).then(function (value) {
             isMounted() && callId === lastCallId.current && set({ value: value, loading: false });
             return value;
         }, function (error) {
             isMounted() && callId === lastCallId.current && set({ error: error, loading: false });
-            return error;
+            throw error;
         });
     }, deps);
     return [state, callback];
diff --git a/lib/useAsync.js b/lib/useAsync.js
index 7f189a49dea552b5b10d7380b982bfe84299a7a2..7d25371b7500006490d2a1f7d414b1eb33307318 100644
--- a/lib/useAsync.js
+++ b/lib/useAsync.js
@@ -9,7 +9,11 @@ function useAsync(fn, deps) {
         loading: true,
     }), state = _a[0], callback = _a[1];
     react_1.useEffect(function () {
-        callback();
+        try {
+            callback();
+        } catch (error) {
+            // do nothing
+        }
     }, [callback]);
     return state;
 }
diff --git a/lib/useAsyncFn.js b/lib/useAsyncFn.js
index e06fd819ccad625d709fa9907e946a9b8bc58543..929e55304b8aef35572046533306a6c35c75a81a 100644
--- a/lib/useAsyncFn.js
+++ b/lib/useAsyncFn.js
@@ -15,15 +15,16 @@ function useAsyncFn(fn, deps, initialState) {
             args[_i] = arguments[_i];
         }
         var callId = ++lastCallId.current;
-        if (!state.loading) {
-            set(function (prevState) { return (tslib_1.__assign(tslib_1.__assign({}, prevState), { loading: true })); });
-        }
+        set(function (prevState) {
+            if (prevState.loading) return prevState
+            return (tslib_1.__assign(tslib_1.__assign({}, prevState), { loading: true })); 
+        });
         return fn.apply(void 0, args).then(function (value) {
             isMounted() && callId === lastCallId.current && set({ value: value, loading: false });
             return value;
         }, function (error) {
             isMounted() && callId === lastCallId.current && set({ error: error, loading: false });
-            return error;
+            throw error;
         });
     }, deps);
     return [state, callback];
