diff --git a/dist/backend.js b/dist/backend.js
index d710b0a5539e368ddb1dace48efbcffdfe91436d..5b5cd65b33fc38dd73303622917b76d03943d802 100644
--- a/dist/backend.js
+++ b/dist/backend.js
@@ -12414,6 +12414,7 @@ function attach(hook, rendererID, renderer, global) {
     updateComponentFilters
   };
 }
+if (sessionStorageGetItem("React::DevTools::reloadAndProfile") === "true") globalThis.__REACT_DEVTOOLS_ATTACH__ = attach;
 ;// CONCATENATED MODULE: ../react-devtools-shared/src/backend/console.js
 /**
  * Copyright (c) Meta Platforms, Inc. and affiliates.
diff --git a/dist/frontend.js b/dist/frontend.js
index 0f3546c887617dc61971eb4715016e5357bcf9e1..702c32367871a5fb05b9d97713bfd2f9d4d37945 100644
--- a/dist/frontend.js
+++ b/dist/frontend.js
@@ -32937,7 +32937,7 @@ columnNumber) {
     if (resourceLine.includes(SOURCE_MAP_ANNOTATION_PREFIX)) {
       const sourceMapAnnotationStartIndex = resourceLine.indexOf(SOURCE_MAP_ANNOTATION_PREFIX);
       const sourceMapURL = resourceLine.slice(sourceMapAnnotationStartIndex + SOURCE_MAP_ANNOTATION_PREFIX.length, resourceLine.length);
-      const sourceMap = await fetchFileWithCaching(sourceMapURL).catch(() => null);
+      const sourceMap = await fetchFileWithCaching(new URL(sourceMapURL, sourceURL).toString()).catch(() => null);
 
       if (sourceMap != null) {
         try {
