diff --git a/lib/optimize/ConcatenatedModule.js b/lib/optimize/ConcatenatedModule.js
index 74b9b18a96f34ba9bbf62a3f6da625d74d3e2177..3fd1667b6473a0ebb76a37c5e090deb91818e3ac 100644
--- a/lib/optimize/ConcatenatedModule.js
+++ b/lib/optimize/ConcatenatedModule.js
@@ -23,6 +23,7 @@ const Template = require("../Template");
 const { DEFAULTS } = require("../config/defaults");
 const HarmonyImportDependency = require("../dependencies/HarmonyImportDependency");
 const HarmonyImportSideEffectDependency = require("../dependencies/HarmonyImportSideEffectDependency");
+const HarmonyImportSpecifierDependency = require("../dependencies/HarmonyImportSpecifierDependency");
 const JavascriptParser = require("../javascript/JavascriptParser");
 const {
 	getMakeDeferredNamespaceModeFromExportsType,
@@ -1831,7 +1832,8 @@ ${defineGetters}`
 						for (const dep of info.module.dependencies) {
 							if (
 								!dep.defer &&
-								dep instanceof HarmonyImportSideEffectDependency
+								(dep instanceof HarmonyImportSpecifierDependency ||
+									dep instanceof HarmonyImportSideEffectDependency)
 							) {
 								const referredModule = moduleGraph.getModule(dep);
 								if (!referredModule) continue;
@@ -1842,7 +1844,13 @@ ${defineGetters}`
 												i.type === "external" && i.module === referredModule
 										)
 									);
-									if (!deferredModuleInfo) continue;
+									if (!deferredModuleInfo) {
+										if (dep instanceof HarmonyImportSideEffectDependency) {
+											continue;
+										} else {
+											throw new Error("webpack deferImport bug");
+										}
+									}
 									result.add(
 										`\n// non-deferred import to a deferred module (${referredModule.readableIdentifier(requestShortener)})\nvar ${deferredModuleInfo.name} = ${deferredModuleInfo.deferredName}.a;`
 									);
